lib/compress.js

```diff
@@ -372,14 +372,16 @@ merge(Compressor.prototype, {
                         //   (function(a,b) {...})(c,d) => (function() {var a=c,b=d; ...})()
                         // So existing transformation rules can work on them.
                         node.argnames.forEach(function(arg, i) {
-                            var d = arg.definition();
-                            if (!node.uses_arguments && d.fixed === undefined) {
-                                d.fixed = function() {
-                                    return iife.args[i] || make_node(AST_Undefined, iife);
-                                };
-                                mark(d, true);
-                            } else {
-                                d.fixed = false;
+                            if (arg.definition instanceof Function) {
```

Comment:
I think `if (arg.definition) {` is sufficient in this case.

The better fix would be to detect the assignment and put the value in accordingly, but that's more work.